pkg/system/utils.go

```diff
@@ -94,3 +94,18 @@ func IsDebugMode() bool {
 	// TODO: #4535 we need to add a flag to the CLI to enable debug mode
 	return os.Getenv("DEBUG") == "true" || zerolog.GlobalLevel() <= zerolog.DebugLevel
 }
+
+// ExtractJSONOutput extracts JSON data from the output
+func ExtractJSONOutput(output string) (jsonData string, remainingOutput string, err error) {
+	start := strings.Index(output, "[")
```

Comment:
I can see you are looking for the first index of the `[` , and the last index of `]`. 
Few comments on that:
1. I know that the "--json" output for this command unfortunately adds extra lines to the end that are not even JSON, so I know the pain and reason to add this logic.
2. Since your function is generic, what if the JSON output is not a JSON array, bit it is a json object ? so it will not start with a `[` .
3. You are adding this extract method to `pkg/system/utils.go`, but only using it in the tests. The name `ExtractJSONOutput` in this common package gives the illusion it is generic, but it not.

My recommendation is to move this helper method to the test file itself since it only applies for this  specific test case. 
This way you do not need to change the logic to support so many edge cases for parsing JSON, and just have the logic to support this test.

Note for later: This is a reminder for us to fix "--json" output for all commands and only produce JSON with no extra data.